samza-api/src/main/java/org/apache/samza/application/StreamApplication.java

```diff
@@ -16,17 +16,20 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.samza.operators;
+package org.apache.samza.application;
 
 import org.apache.samza.annotation.InterfaceStability;
 import org.apache.samza.config.Config;
+import org.apache.samza.operators.StreamGraph;
 
 
 /**
  * This interface defines a factory class that user will implement to create user-defined operator DAG in a {@link StreamGraph} object.
  */
 @InterfaceStability.Unstable
-public interface StreamGraphBuilder {
+public interface StreamApplication {
+  static final String APP_CLASS_CONFIG = "app.class";
```

Comment:
Minor: This should probably not be here. I think so far we've kept it in config classes. If there isn't one, maybe keep in TaskFactoryUtils?